pyxpp/ast_builder/ast_node.py

```diff
@@ -0,0 +1,147 @@
+from typing import List, Optional, Union
+
+
+# Базовый класс для всех узлов AST
+class ASTNode:
+    """Base class for all AST nodes."""
+
+    def to_string(self) -> str:
+        """Returns a string representation of the node. Override in subclasses."""
+        return "ASTNode"
+    
+class NodeWithChildren:
+    """Base class for all AST nodes with children."""
+    def __init__(self):
+        self.children = []
+
+    def add_child(self, child: 'ASTNode') -> None:
+        """Adds a child to the node and sets its parent."""
+        self.children.append(child)
+
+    def to_string(self) -> str:
+        """Returns a string representation of the node. Override in subclasses."""
+        return "ASTNode"
+    
+
+class Program(ASTNode, NodeWithChildren):
```

Comment:
I don't see the point of having a separate node class and a node with descendants.
Simply put, it's best to combine ASTNode and NodeWithChildreninto a single ASTNode class.
It's okay if nodes with descendants don't have descendants.

Since there is no need to define the to_string method, because there is \_\_str\_\_

Reply:
done.